src/Raven.Server/Documents/TransactionOperationsMerger.cs

```diff
@@ -110,6 +145,164 @@ private static void ThrowTxMergerWasDisposed()
             throw new ObjectDisposedException("Transaction Merger");
         }
 
+
+        public abstract class RecordingState : IDisposable
+        {
+            public abstract void Record(MergedTransactionCommand cmd);
+
+            public abstract void Record(TxInstruction tx, bool doRecord = true);
+
+            public abstract void Prepare(ref RecordingState state);
+
+            protected class EnabledRecordingState : RecordingState
+            {
+                private readonly TransactionOperationsMerger _txOpMerger;
+                private readonly Stream _recordingFileStream;
+                private bool isDisposed = false;
+
+                public EnabledRecordingState(TransactionOperationsMerger txOpMerger, Stream recordingFileStream)
+                {
+                    _txOpMerger = txOpMerger;
+                    _recordingFileStream = recordingFileStream;
+                }
+
+                public override void Record(MergedTransactionCommand operation)
+                {
+                    var obj = new DynamicJsonValue
+                    {
+                        [TypeKey] = operation.GetType().Name,
+                    };
+
+                    if ((operation is IRecordableCommand recordable))
+                    {
+                        var recOpt = recordable.Serialize();
+                        obj[CommandKey] = recOpt;
+                    }
+
+                    Record(obj);
+                }
+
+                public override void Record(TxInstruction tx, bool doRecord = true)
+                {
+                    if (doRecord == false)
+                    {
+                        return;
+                    }
+
+                    Record(new DynamicJsonValue
+                    {
+                        [TypeKey] = tx.ToString(),
+                    });
+                }
+
+                private void Record(DynamicJsonValue obj)
+                {
+                    obj[DateTimeKey] = DateTime.Now;
+
+                    using (_txOpMerger._parent.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext ctx))
+                    {
+                        using (var writer = new BlittableJsonTextWriter(ctx, _recordingFileStream))
+                        {
+                            writer.WriteComma();
```

Comment:
Is this comma correct?

Reply:
Yes, there is always a first default value